bundles/org.openhab.binding.kaleidescape/src/main/java/org/openhab/binding/kaleidescape/internal/communication/KaleidescapeSerialConnector.java

```diff
@@ -0,0 +1,132 @@
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.io.UnsupportedEncodingException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.io.transport.serial.PortInUseException;
+import org.eclipse.smarthome.io.transport.serial.SerialPort;
+import org.eclipse.smarthome.io.transport.serial.SerialPortIdentifier;
+import org.eclipse.smarthome.io.transport.serial.SerialPortManager;
+import org.eclipse.smarthome.io.transport.serial.UnsupportedCommOperationException;
+import org.openhab.binding.kaleidescape.internal.KaleidescapeException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Class for communicating with the Kaleidescape component through a serial connection
+ *
+ * @author Laurent Garnier - Initial contribution
+ * @author Michael Lobstein - Adapted for the Kaleidescape binding
+ */
+@NonNullByDefault
+public class KaleidescapeSerialConnector extends KaleidescapeConnector {
+    private final Logger logger = LoggerFactory.getLogger(KaleidescapeSerialConnector.class);
+
+    private final String serialPortName;
+    private final SerialPortManager serialPortManager;
+    private final String uid;
+
+    private @Nullable SerialPort serialPort;
+
+    /**
+     * Constructor
+     *
+     * @param serialPortManager the serial port manager
+     * @param serialPortName the serial port name to be used
+     * @param uid the thing uid string
+     */
+    public KaleidescapeSerialConnector(SerialPortManager serialPortManager, String serialPortName, String uid) {
+        this.serialPortManager = serialPortManager;
+        this.serialPortName = serialPortName;
+        this.uid = uid;
+    }
+
+    @Override
+    public synchronized void open() throws KaleidescapeException {
+        logger.debug("Opening serial connection on port {}", serialPortName);
+        try {
+            SerialPortIdentifier portIdentifier = serialPortManager.getIdentifier(serialPortName);
+            if (portIdentifier == null) {
+                setConnected(false);
+                throw new KaleidescapeException("Opening serial connection failed: No Such Port");
+            }
+
+            SerialPort commPort = portIdentifier.open(this.getClass().getName(), 2000);
+
+            commPort.setSerialPortParams(19200, SerialPort.DATABITS_8, SerialPort.STOPBITS_1, SerialPort.PARITY_NONE);
+            commPort.enableReceiveThreshold(1);
+            commPort.enableReceiveTimeout(100);
+            commPort.setFlowControlMode(SerialPort.FLOWCONTROL_NONE);
+
+            InputStream dataIn = commPort.getInputStream();
+            OutputStream dataOut = commPort.getOutputStream();
+
+            if (dataOut != null) {
+                dataOut.flush();
+            }
+            if (dataIn != null && dataIn.markSupported()) {
+                try {
+                    dataIn.reset();
+                } catch (IOException e) {
+                }
```

Comment:
You should still log something here at the very least.

Reply:
done